tests/layer_tests/tensorflow_tests/test_tf_MatMul.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestMatMul(CommonTFLayerTest):
+
+    def create_net_with_matmul_op(self, x_shape, y_shape, x_bool, y_bool, op_type, ir_version, use_new_frontend):
+        """
+            Tensorflow net                 IR net
+
+            Input->op_type        =>       Input->MatMul
+                     /                              /
+            Const---/                      Const---/
+        """
+        import tensorflow as tf
+        op_type_to_tf = {
+            'BatchMatMul': tf.raw_ops.BatchMatMul,
+            'BatchMatMulV2': tf.raw_ops.BatchMatMulV2,
+            'BatchMatMulV3': tf.raw_ops.BatchMatMulV3,
+            'MatMul': tf.raw_ops.MatMul,
+        }
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            tf_x_shape = x_shape.copy()
+            tf_y_shape = y_shape.copy()
+
+            tf_x = tf.compat.v1.placeholder(tf.float32, tf_x_shape, 'Input')
+            constant_value = np.random.randint(-3, 3, tf_x_shape).astype(np.float32)
+            tf_y = tf.constant(constant_value)
```

Comment:
let us have both placeholders instead of one placeholder and one constant

Reply:
np